kvdb/sqlite/db.go

```diff
@@ -50,6 +54,18 @@ func NewSqliteBackend(ctx context.Context, cfg *Config, dbPath, fileName,
 			value: "WAL",
 		},
 	}
+
+	// If the database is being created, set auto_vacuum to incremental.
+	// This must be done before any tables are created. According to the
+	// SQLite docs, this corresponds to mode 2.
+	if isCreate {
+		// Use the new named struct pragmaOption here as well.
+		pragmaOptions = append(pragmaOptions, pragmaOption{
```

Comment:
Note that to make it active, we'll also need an additional pragma when opening the DB.

From the docs:
"When the value of auto-vacuum is 2 or "incremental" then the additional information needed to do auto-vacuuming is stored in the database file but auto-vacuuming does not occur automatically at each commit as it does with auto_vacuum=full. In incremental mode, the separate [incremental_vacuum](https://sqlite.org/pragma.html#pragma_incremental_vacuum) pragma must be invoked to cause the auto-vacuum to occur."

ref: https://sqlite.org/pragma.html#pragma_auto_vacuum

Reply:
Yeah I view that as a follow up set. We'd add a background goroutine that calls `incremental_vacuum` in the background. 

We do this now in isolation as you can't add this pragma once the DB has already been created.

Reply:
sgtm